### PR TITLE
Chasm liquid drain

### PR DESCRIPTION
**Pull request by Tyzemol**
## About the PR

Makes chasms remove liquids on the same tile.
image

## Why / Balance

Spite PR after i got roundremoved on an expedition due to a huge puddle completely hiding the chasm.

## Technical details

Adds drain component to chasms.
Adds a bool toggle to disable examinable available volume in drains so you cant see how much volume a chasm has.
## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

:cl:
- tweak: Liquids can no longer magically hover over bottomless chasms

### CODE REVIEW

**Comment by whateverusername0** [comment]
epic chasm death trap fail.😢

merge now

File: Content.Shared/Fluids/Components/DrainComponent.cs [hunk 1]
```diff
@@ -33,6 +33,12 @@ public sealed partial class DrainComponent : Component
     [DataField("autoDrain"), ViewVariables(VVAccess.ReadOnly)]
     public bool AutoDrain = true;
 
+    /// <summary>
+    /// Does examining this drain show available volume?
+    /// </summary>
+    [DataField("examineAvailableVolume"), ViewVariables(VVAccess.ReadOnly)]
```
Comment (Tayrtahn) [inline Content.Shared/Fluids/Components/DrainComponent.cs:39]:
```suggestion
    [DataField, ViewVariables(VVAccess.ReadOnly)]
```
Not sure why this needs to be read only either.

**Comment by TheShuEd** [comment]
I suggest adding this same functionality to rivers, magma and liquid plasma

**Comment by Errant-4** [comment]
And lattices in space

**Comment by DogZeroX** [comment]
and space

**Comment by TheShuEd** [comment]
> And lattices in space

This is tiles, not entities, so i think its out of scope

**Comment by Errant-4** [comment]
> and space

Removing a tile entirely and leaving just space does seem to already immediately destroy the puddle on it

**Comment by EmoGarbage404** [comment]
drains are kinda expensive in terms of performance so i dont really know if sticking entities that are placed as regularly in large groups as chasms, water, lava, etc. is okay in terms of performance.

**Comment by deltanedas** [comment]
wouldnt it be more feasible to have spreaders not be allowed to spread over chasms etc in the first place

File: Content.Shared/Fluids/Components/DrainComponent.cs (hunk 1 above)
Comment (TheShuEd) [reply Content.Shared/Fluids/Components/DrainComponent.cs:39]: this shouldn't exist, we have an ExaminableSolutionComponent

**Review by TheShuEd — CHANGES_REQUESTED** [review]
Yes, this will definitely require a different implementation. It will be too high load on the server.

**Comment by slarticodefast** [comment]
Are you coming back to this? It will need a different approach that is better performance-wise though.

**Comment by metalgearsloth** [comment]
You would essentially need to hook into spreader code because running 100s of drains would bring down a server.

**Comment by slarticodefast** [comment]
Please write a comment if you come back to this and I will re-open the PR.

**Pull request closed without merging**